arch/arm/core/swap_helper.S

```diff
@@ -560,9 +560,7 @@ demux:
     beq _oops
 
 #if CONFIG_IRQ_OFFLOAD
-    push {r0, lr}
```

Comment:
So the LR value at this point is never needed in the SCV handler routine?

Reply:
I am not really fond of skipping the push/pop sequence around a BL instruction, this seems prone to errors.
If you still want to do that, I think you probably need to write down the full argument as comment, that is, _why_ it is not needed to store the LR. 

/* exception return is done in _IntExit() */ does not reveal the whole story

Reply:
> I am not really fond of skipping the push/pop sequence around a BL instruction, this seems prone to errors.

As long as lr is preserved (i.e. lr at the time of entering and exiting ISR is the same), there should be no problem. In fact, isr_wrapper is already doing exactly what I intend to do here:
https://github.com/zephyrproject-rtos/zephyr/blob/8c4376e29000cdfbbbe781a40c3977e30dddac0d/arch/arm/core/isr_wrapper.S#L63-L69

Reply:
> So the LR value at this point is never needed in the SCV handler routine?

For Cortex-R, __svc saves the value of lr into the system mode stack and _IntExit restores this from the system stack into lr, before returning from SVC or IRQ mode.

https://github.com/zephyrproject-rtos/zephyr/blob/8c4376e29000cdfbbbe781a40c3977e30dddac0d/arch/arm/core/swap_helper.S#L519-L534

Reply:
Exactly. Could you add some inline documentation for that?

Reply:
> Exactly. Could you add some inline documentation for that?

Add a new commit addressing this: https://github.com/zephyrproject-rtos/zephyr/pull/19389/commits/bc50dffd06699ab17353b8e73c723209c112ea3e